src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -372,14 +372,21 @@ public function getGlobalScopes()
 	/**
 	 * Register an observer with the Model.
 	 *
-	 * @param  object  $class
+	 * @param  object|string $class
 	 * @return void
 	 */
 	public static function observe($class)
 	{
 		$instance = new static;
 
-		$className = get_class($class);
+		if (is_string($class))
```

Comment:
Yeah, wasn’t sure if you had a preference one way or the other, so I left it as it was in the original PR. I’ll switch that and rebase onto a single commit in just a minute

-- 
Mike Dugan
mike@dugan.io
http://dugan.io

On February 27, 2015 at 9:33:54 AM, Taylor Otwell (notifications@github.com) wrote:

In src/Illuminate/Database/Eloquent/Model.php:

>         * @return void
>     */
>    public static function observe($class)
>    {
>            $instance = new static;
>   
> -          $className = get_class($class);
> +          if (is_string($class))
This could just use the ternary operator.

—
Reply to this email directly or view it on GitHub.